Rs fixes first build mac os issues

xn-zhao opened the pull request:
fixes all build issues on Mac OSx.

vivo010 committed:
Adding Mac OSX Compatibility compilation fixes.

(cherry picked from commit 9096a927e297de8cc80968b6189da36da1f9214c)

vivo010 committed:
Adding Mac OSX bdboot.txt fixed

xn-zhao committed:
Fixes all build issues on Mac OS.

xn-zhao committed:
comment for all code customization of fix first build issues on Mac

csoler commented on RetroShare.pro:
```diff
@@ -17,7 +17,9 @@
-
+#atai add this one
+CONFIG += qt gui uic qrc resources uitools debug
+CONFIG+=rs_async_chat
```
why do you need this?

xn-zhao replied:
Thank you for your comment.
The first one I think we can remove it. I just add for debug build, but we can set it in another place.
The second one is the one that I think it can send msg in the distant chat (in people tab) without establishing the secure tunnel, but it still not work at this time, so we can remove it too. I just try to commit to the one master of one account for first test, but it 'touchs' to all our community. So sorry.

csoler commented on RetroShare.pro:
```diff
@@ -17,7 +17,9 @@
 !include("retroshare.pri"): error("Could not include file retroshare.pri")
-
+#atai add this one
+CONFIG += qt gui uic qrc resources uitools debug
```
I dont think this is needed. libretroshare that is below this directory does not need qt. All this should be in retroshare-gui/src/retroshare-gui.pro, and it's probably already there.

csoler commented on libretroshare/src/chat/rschatitems.cc:
```diff
@@ -29,7 +29,7 @@
 #include "chat/rschatitems.h"
 
-//#define CHAT_DEBUG 1
+#define CHAT_DEBUG 1
```
you forgot to leave this commented out.

xn-zhao replied:
Thanks for this

csoler commented on libretroshare/src/util/cxx11retrocompat.h:
```diff
@@ -19,15 +19,15 @@
  *                                                                             *
  *******************************************************************************/
-#pragma once
+//#pragma once
```
#pragma once has no reason to be commented out. Why would you need to do that?

Commenting the entire bloc out below seems dangerous to me given that is has been put here to allow to compile on other systems. At least put a proper #ifndef MACOS around.

xn-zhao replied:
Yes, I think in some places we will put the platform checking, so that it will work for all. This one is the case.

G10h4ck replied:
`#pragma once` should be supported by any decent compiler out there

csoler commented on openpgpsdk/src/openpgpsdk.pro:
```diff
@@ -2,7 +2,8 @@
 CONFIG += staticlib
-
+## Sep 11 2018 - meiyousixin - fix bug: can not find openssl/bn.h
+INCLUDEPATH += /usr/local/include
```
that should be set in the .pri I suppose.

xn-zhao replied:
ok, i will move it to .pri, thanks.

csoler commented on retroshare.pri:
```diff
@@ -118,9 +118,11 @@ CONFIG *= rs_gxs_trans
 # command line "CONFIG+=rs_async_chat"
-CONFIG *= no_rs_async_chat
+#CONFIG *= no_rs_async_chat
+CONFIG+=rs_async_chat
```
rs_async_chat should be left disabled by default. How can adding additional code help compilation anyway?

csoler commented:
thx for this PR. Your fixes give some clues about what to do to fix compilation, but there's lots of questions to solve before we merge this. Please see the inlined comments.

vivo010 committed:
Adding Mac OSX Macro to fix building issues.

vivo010 committed:
Adding Mac OSX Configurate and compatibility

vivo010 committed:
Adding Mac OSX Configuration and compatibility fixes

vivo010 committed:
Merge branch 'master' of https://github.com/RetroShare/RetroShare

Sync up with Retroshare Master branch

chinhniem80 committed:
Move contact list from network to contact tab

bducduy committed:
Update images in chats and app

chinhniem80 committed:
update retroshare.pri: rs_async_chat

bducduy committed:
Uptdate logo unseen

phikhong committed:
Update logo and background unseen

phikhong committed:
update logo splash

phikhong committed:
Update icon unseen

phikhong committed:
update icon

phikhong committed:
update GenCertDialog.ui

phikhong committed:
replace content in HelpButton

phikhong committed:
Update avatar and icon P2PUnseen

phikhong committed:
update color

phikhong committed:
change color of tabs: person and circles. change toolBarArea position

phikhong committed:
set QToolBar color

phikhong committed:
update P2Unseen (in toaster file)

phikhong committed:
Revert "update P2Unseen (in toaster file)"

This reverts commit 68992c096d2063f3827a2cb7fd6cb59d2a9ea8da.

phikhong committed:
Revert "Revert "update color""

This reverts commit fa920f28b931f2fe07f264044ff4ba39a5f64d65.

phikhong committed:
update P2Unseen (in toaster file)

phikhong committed:
update unseen logo size 16

phikhong committed:
replace P2PUnseen text in .cpp file

phikhong committed:
replace P2PUnseen text in .cpp file (2)

phikhong committed:
replace P2PUnseen text in .ui file

chinhniem80 committed:
8.1. Make UI for One2One Chat : Add one2one chat into chats tab with groupchat

chinhniem80 committed:
merge UI of Duy from Test2-d

unsenep2pdev committed:
Merge pull request #3 from unsenep2pdev/rs-fixes-first-build-MacOS-issues

Add Unseen P2P Layout & GUI

unsenep2pdev committed:
Update openpgpsdk.pro

changing openssl path to local mac osx issue
	INCLUDEPATH += /usr/local/include

unsenep2pdev committed:
Update retroshare.pri

Removing QT BIN PATH         BIN_DIR += "/Users/username/Qt/5.9/clang_64/bin", however need to specify QT_HOME_DIR, will need to setup with your specific environment.

unsenep2pdev committed:
Merge pull request #4 from unsenep2pdev/rs-fixes-first-build-MacOS-issues

Local environment for mac osx remove

unsenep2pdev committed:
Update README.md

phikhong committed:
Update Unseen avatar of friends in Person tab

phikhong committed:
Update Unseen avatar of friends in Person tab

phikhong committed:
Update Unseen default avatar

phikhong committed:
Update Unseen default avatar

vivo010 committed:
Fixing the Crash bugs

phikhong committed:
Set toolbar

phikhong committed:
Set toolbar

phikhong committed:
Update interface

phikhong committed:
Update interface

chinhniem80 committed:
change UI logics: remove channels, links, people, logs tabs, using network as contact tab

chinhniem80 committed:
remove all comments

phikhong committed:
update ui: toolbar and icon

vivo010 committed:
Temp fix DHT invalid IP format crash on Mac OSX.

vivo010 committed:
we adjust debug text to align with original debug option.

unsenep2pdev committed:
Update .travis.yml

unsenep2pdev committed:
restore to original cross compilation setting.

comment out Local Mac OSX Environment Setting.

unsenep2pdev committed:
Update retroshare.pri

unsenep2pdev committed:
Update retroshare.pri

unsenep2pdev committed:
Update .travis.yml

adding libgnome-keyring-dev for ubuntu platform to build enable save login version.

unsenep2pdev committed:
Update .travis.yml

unsenep2pdev committed:
Update .travis.yml

unsenep2pdev committed:
Update .travis.yml

unsenep2pdev committed:
Update .travis.yml

unsenep2pdev committed:
Update .travis.yml

unsenep2pdev committed:
Update .travis.yml

unsenep2pdev committed:
Update .travis.yml

unsenep2pdev committed:
Update .travis.yml

unsenep2pdev committed:
Update cxx11retrocompat.h

unsenep2pdev committed:
Update cxx11retrocompat.h

unsenep2pdev committed:
Update .travis.yml

chinhniem80 committed:
remove debug define and fix Qt version gettime funtcion on travis

unsenep2pdev committed:
Update .travis.yml

unsenep2pdev committed:
Update retroshare.pri

unsenep2pdev committed:
Update .travis.yml

unsenep2pdev committed:
Update libretroshare.pro

unsenep2pdev committed:
Update libretroshare.pro

unsenep2pdev committed:
Update retroshare.pri

unsenep2pdev committed:
Update .travis.yml

unsenep2pdev committed:
Update retroshare.pri

unsenep2pdev committed:
Update .travis.yml

chinhniem80 committed:
merge with master

chinhniem80 committed:
update retroshare.pri as master

chinhniem80 committed:
fix Qt version for QDatetime:currentSecsSinceEpoch function

chinhniem80 committed:
fix QDateTime::currentDateTime()

csoler commented:
Sorry guys, but a PR that changes 899 files including global replacement of "retroshare" for "p2punseen" does not make sense. If you really want to contribute, please make small pull requests with sensible changes.
